tabular/src/autogluon/tabular/models/catboost/catboost_model.py

```diff
@@ -280,3 +280,11 @@ def get_model_feature_importance(self):
         importance_series = importance_df.set_index('Feature Id')['Importances']
         importance_dict = importance_series.to_dict()
         return importance_dict
+
+    def _get_default_auxiliary_params(self) -> dict:
+        default_auxiliary_params = super()._get_default_auxiliary_params()
+        extra_auxiliary_params = dict(
+            ignored_type_group_raw=[R_OBJECT],
+        )
+        default_auxiliary_params.update(extra_auxiliary_params)
```

Comment:
If the merge approach is the expected pattern of usage to add additional parameters, this would be error-prone: please consider generalizing the merge logic in the abstract class and allow extension to return just `extra_auxiliary_params`.

Reply:
[Discussion point] Also, does it makes sense to force sub-classes to implement this method? If so, then abstract class can throw a `NotImplementedError` by default. This would guide model author to override all the necessary methods.

Reply:
How would this work if the model extended a class which then extended AbstractModel? We would need to handle chained calls to extra_auxiliary_params.

Reply:
Yes, the class hierarchy would make it more complicated. I think this alternative would work better: perform overwrites in `__init__`, instead - this would preserve hierarchy chain overwrites:
```python
class AbstractModel(object):

    def __init__(self):
        self.auxiliary_params = dict(
            max_memory_usage_ratio=1.0,  # Ratio of memory usage allowed by the model. Values > 1.0 have an increased risk of causing OOM errors. Used in memory checks during model training to avoid OOM errors.
            max_time_limit_ratio=1.0,  # ratio of given time_limit to use during fit(). If time_limit == 10 and max_time_limit_ratio=0.3, time_limit would be changed to 3.
        )

    def register_auxiliary_params(self, extra_auxiliary_params: dict):
        self.auxiliary_params.update(extra_auxiliary_params)


class SubModel(AbstractModel):

    def __init__(self):
        super().__init__()
        self.register_auxiliary_params(dict(
            max_time_limit_ratio=2.0,
            new_param='a',
        ))


class SubSubModel(SubModel):

    def __init__(self):
        super().__init__()
        self.register_auxiliary_params(dict(
            max_memory_usage_ratio=3.0,
            new_sub_param='c',
        ))

if __name__ == '__main__':
    print(f'AbstractModel: {AbstractModel().auxiliary_params}')
    print(f'SubModel     : {SubModel().auxiliary_params}')
    print(f'SubSubModel  : {SubSubModel().auxiliary_params}')
```
Output:
```
AbstractModel: {'max_memory_usage_ratio': 1.0, 'max_time_limit_ratio': 1.0}
SubModel     : {'max_memory_usage_ratio': 1.0, 'max_time_limit_ratio': 2.0, 'new_param': 'a'}
SubSubModel  : {'max_memory_usage_ratio': 3.0, 'max_time_limit_ratio': 2.0, 'new_param': 'a', 'new_sub_param': 'c'}
```

Reply:
The problem comes that the user can override the defaults by the params argument. This can't be done properly if done in init since this logic happens in abstract_model's init, and thus this order will overwrite user specified overrides.